### PR TITLE
fix non sudo enable community addon

### LINKED ISSUE #4574: Error enabling community addon in MicroK8s 1.30.1
#### Summary
I encountered an error when trying to enable the community addon in MicroK8s version 1.30.1. The error message suggests an issue with repository ownership and permissions. This issue did not occur in MicroK8s version 1.29, where the same command worked fine.

#### What Should Happen Instead?
The community addon should be enabled without any errors, as it did in version 1.29.

#### Reproduction Steps

1. Install MicroK8s version 1.30.1
2. Run the command `microk8s enable community`

#### Error Log
```
Infer repository core for addon community
Cloning into '/var/snap/microk8s/common/addons/community'...
fatal: detected dubious ownership in repository at '/snap/microk8s/current/addons/community/.git'
To add an exception for this directory, call:

git config --global --add safe.directory /snap/microk8s/current/addons/community/.git

fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Traceback (most recent call last):
File "/snap/microk8s/6876/scripts/wrappers/addons.py", line 351, in <module>
addons(prog_name="microk8s addons")
File "/snap/microk8s/6876/usr/lib/python3/dist-packages/click/core.py", line 764, in **call**
return self.main(*args, **kwargs)
File "/snap/microk8s/6876/usr/lib/python3/dist-packages/click/core.py", line 717, in main
rv = self.invoke(ctx)
File "/snap/microk8s/6876/usr/lib/python3/dist-packages/click/core.py", line 1137, in invoke
return _process_result(sub_ctx.command.invoke(sub_ctx))
File "/snap/microk8s/6876/usr/lib/python3/dist-packages/click/core.py", line 1137, in invoke
return _process_result(sub_ctx.command.invoke(sub_ctx))
File "/snap/microk8s/6876/usr/lib/python3/dist-packages/click/core.py", line 956, in invoke
return ctx.invoke(self.callback, **ctx.params)
File "/snap/microk8s/6876/usr/lib/python3/dist-packages/click/core.py", line 555, in invoke
return callback(*args, **kwargs)
File "/snap/microk8s/6876/scripts/wrappers/addons.py", line 207, in add
subprocess.check_call(cmd)
File "/snap/microk8s/6876/usr/lib/python3.8/subprocess.py", line 364, in check_call
raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/snap/microk8s/6876/git.wrapper', 'clone', '/snap/microk8s/current/addons/community', PosixPath('/var/snap/microk8s/common/addons/community')]' returned non-zero exit status 128.
```

Thank you 😊

### PR DESCRIPTION
**Pull request by maci3jka**
#### Summary
This PR fixes issues when non-root user in non-strict environment tries to enable community addons.

Closes #4574

#### Changes
added temporary setting to git to  ingnore

#### Testing
enabled and disabled community addon

#### Possible Regressions
n/a

#### Checklist

* [x] Read the [contributions](https://github.com/canonical/microk8s/blob/master/CONTRIBUTING.md) page.
* [x] Submitted the [CLA form](https://ubuntu.com/legal/contributors/agreement), if you are a first time contributor.
* [ ] The introduced changes are covered by unit and/or integration tests.

### CODE REVIEW

**Review by berkayoz — COMMENTED** [review]
I'm not too happy about this workaround, this feels like just masking the underlying issue. The safe directory wildcard also feels too wide, could we deem certain directories as safe instead of any?

**Comment by maci3jka** [comment]
> I'm not too happy about this workaround, this feels like just masking the underlying issue. The safe directory wildcard also feels too wide, could we deem certain directories as safe instead of any?

How about narowing it to ` $SNAP/addons/community/*` ?

File: microk8s-resources/gitconfig [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[safe]
```
Comment (berkayoz) [inline microk8s-resources/gitconfig:1]: Can we put this like `microk8s-resources/default-args/git/.gitconfig`?

File: microk8s-resources/wrappers/git.wrapper [hunk 2]
```diff
@@ -8,4 +8,6 @@ export GIT_EXEC_PATH="$SNAP/usr/lib/git-core"
 export GIT_TEMPLATE_DIR="$SNAP/usr/share/git-core/templates"
 export GIT_CONFIG_NOSYSTEM=1
 
+HOME="${SNAP_COMMON}"
```
Comment (berkayoz) [inline microk8s-resources/wrappers/git.wrapper:11]:
```suggestion
HOME="${SNAP_DATA}/args/git"
```

Let's put this under the `args` folder.

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -250,6 +250,7 @@ parts:
       cp client.config.template "${SNAPCRAFT_PART_INSTALL}/client.config.template"
       cp kubelet.config.template "${SNAPCRAFT_PART_INSTALL}/kubelet.config.template"
       cp client-x509.config.template "${SNAPCRAFT_PART_INSTALL}/client-x509.config.template"
+      cp gitconfig "${SNAPCRAFT_PART_INSTALL}/gitconfig"
```
Comment (berkayoz) [inline snap/snapcraft.yaml:253]: We would not need this since `default-args` is copied above.

File: snap/hooks/install [hunk 4]
```diff
@@ -148,4 +148,7 @@ then
   touch ${SNAP_DATA}/var/lock/low-memory-guard.lock
 fi
 
+# copy git config
+cp ${SNAP}/gitconfig  ${SNAP_COMMON}/.gitconfig
+
```
Comment (berkayoz) [inline snap/hooks/install:153]:
```suggestion
# copy git config
mkdir -p ${SNAP_DATA}/args/git
cp ${SNAP}/default-args/git/.gitconfig  ${SNAP_DATA}/args/git/.gitconfig
```

Let's put this under `/var/snap/microk8s/current/args/git`.

**Review by berkayoz — COMMENTED** [review]
Great work, almost there!

File: microk8s-resources/gitconfig (hunk 1 above)
Comment (maci3jka) [reply microk8s-resources/gitconfig:1]: do we want to have it in the repo as a hidden file?

**Review by berkayoz — APPROVED** [review]
LGTM, let's do the trivy fix as a separate card as it still seems to fail here.

**Pull request merged**